utils/record/record.go

```diff
@@ -278,16 +278,28 @@ func (e *EventRecorderAdapter) sendNotifications(object runtime.Object, opts Eve
 		return err
 	}
 
-	res, err := notificationsAPI.RunTrigger(trigger, objMap)
-	if err != nil {
-		log.Errorf("Failed to execute condition of trigger %s: %v", trigger, err)
-		return err
-	}
-	log.Infof("Trigger %s result: %v", trigger, res)
-
-	for _, dest := range destinations {
-		for _, c := range res {
-			if c.Triggered == true {
+	for _, ta := range triggerActions {
+		if ta.When != "" {
+			res, err := notificationsAPI.RunTrigger(trigger, objMap)
+			if err != nil {
+				log.Errorf("Failed to execute condition of trigger %s: %v", trigger, err)
+				return err
+			}
+			log.Infof("Trigger %s result: %v", trigger, res)
+
+			for _, dest := range destinations {
+				for _, c := range res {
+					if c.Triggered == true {
+						err = notificationsAPI.Send(objMap, triggerActions[0].Send, dest)
+						if err != nil {
+							log.Errorf("notification error: %s", err.Error())
+							return err
+						}
+					}
+				}
+			}
+		} else {
+			for _, dest := range destinations {
 				err = notificationsAPI.Send(objMap, triggerActions[0].Send, dest)
```

Comment:
Why are we always sending triggerAction[0] in this loop of trigger actions?

Reply:
not necessary we can ta.. but from the event notification design it is always going to have only once event for which the trigger runs at any point so it would work. However it would be good to replace it with the particular triggeraction at that index as we are anyway looping through the list will update the pr.